internal/pkg/sessions/session_state.go

```diff
@@ -45,6 +46,14 @@ func (s *SessionState) ValidationPeriodExpired() bool {
 	return isExpired(s.ValidDeadline)
 }
 
+// IsWithinGracePeriod returns true if the session is still within the grace period
+func (s *SessionState) IsWithinGracePeriod(gracePeriodTTL time.Duration) bool {
```

Comment:
This is a good refactoring. This PR points out how we could in a followup could/should convert this so that this Period acts like our other periods. 💯